common/djangoapps/student/models.py

```diff
@@ -250,7 +250,7 @@ class Meta:  # pylint: disable=missing-docstring
     country = CountryField(blank=True, null=True)
     goals = models.TextField(blank=True, null=True)
     allow_certificate = models.BooleanField(default=1)
-    bio = models.TextField(blank=True, null=True)
+    bio = models.CharField(blank=True, max_length=3000, db_index=False)
```

Comment:
hm.   well, `null=True` definitely matters to the physical model that's generated in the database - the second answer to the SO question you linked above contains a test that proves this.  the main thing is that it must be set the same in the model definition and migration files, since otherwise there's a risk of south getting confused.